### DISCUSSION TITLE
Does ukv support TTL?

### DISCUSSION THREAD

**Discussion by kesavkolla** [unum-cloud/UStore#230] [Q&A] [answered]
Can we set expirations for keys? This is also very useful feature for key/value store. There is an usecase to store session data in key/value store TTL is useful for eviction of this session data.

**Comment by ashvardanian** [comment, marked as answer]
Not yet, but we are considering this feature for future releases. It is tricky to implement for some modalities, like Graphs. Our more significant priorities are replication and horizontal scaling, but we can prioritize TTL if there is sufficient user demand.

**Comment by kesavkolla** [comment]
If there is no TTL then it's on caller's thing to make sure proper eviction of keys happen. I think it's a good thing to support as part of ukv.

**Reply by Goutamxd** [reply]
True:)
Without TTL, manual eviction risks memory bloat. Built-in TTL in UKV would automate cleanup, boosting efficiency and scalability.🚀
